common/src/primitives/time/rpc.rs

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) 2023 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::{str::FromStr, time::Duration};
+
+use serde::{Deserialize, Serialize};
+
+type RpcDurationInternal = f64;
+
+/// Newtype for time duration, expressed as a floating point number of seconds
+#[derive(Copy, Clone, Debug, Serialize, Deserialize, PartialEq)]
+pub struct RpcDuration(RpcDurationInternal);
+
+#[derive(Debug, thiserror::Error)]
+pub enum RpcTimeError {
+    #[error("Invalid value")]
```

Comment:
Please print the value in the error. Otherwise in a chain of errors it'll be opaque.

Reply:
Removed.